### PR TITLE
Extend BaseQuery with the by_searchable_text method.

### PR DESCRIPTION
**Pull request by phgross**
Used in https://github.com/4teamwork/opengever.core/pull/772.

@lukasgraf @deiferni please have a look ...

### CODE REVIEW

File: opengever/ogds/models/query.py [hunk 1]
```diff
@@ -11,3 +14,28 @@ def _attribute(self, name):
...
+        config's `filter_text` is set.
+        """
+        query = self
+
+        if text and len(text):
+            if isinstance(text, str):
+                text = text.decode('utf-8')
+
+            # remove trailing asterisk
+            if text.endswith('*'):
```
Comment (deiferni) [inline opengever/ogds/models/query.py]: please add a test with asterisk

File: opengever/ogds/models/query.py [hunk 2]
```diff
@@ -11,3 +14,28 @@ def _attribute(self, name):
...
+                text = text[:-1]
+
+            # lets split up the search term into words, extend them with
+            # the default wildcards and then search for every word
+            # seperately
+            for word in text.strip().split(' '):
```
Comment (deiferni) [inline opengever/ogds/models/query.py]: please add a test with multiple search terms

File: opengever/ogds/models/query.py [hunk 3]
```diff
@@ -11,3 +14,28 @@ def _attribute(self, name):
 
         """
         return _entity_descriptor(self._joinpoint_zero(), name)
+
+    def by_searchable_text(self, text=''):
```
Comment (deiferni) [inline opengever/ogds/models/query.py]: hm i wonder - maybe we should move input sanitation to the controller and let this method accept a list of search terms. what do you think?

File: opengever/ogds/models/query.py [hunk 4]
```diff
@@ -11,3 +14,28 @@ def _attribute(self, name):
...
+
+            # remove trailing asterisk
+            if text.endswith('*'):
+                text = text[:-1]
+
+            # lets split up the search term into words, extend them with
+            # the default wildcards and then search for every word
+            # seperately
+            for word in text.strip().split(' '):
+                term = '%%%s%%' % word
```
Comment (deiferni) [inline opengever/ogds/models/query.py]: please use `.format`

**Comment by phgross** [comment]
@deiferni updated please have another look ...

File: opengever/ogds/models/tests/test_query.py [hunk 5]
```diff
@@ -34,3 +34,57 @@ def test_get_one(self):
...
+            User.query.by_searchable_text(['Br', 'example']).all())
+
+        self.assertEqual(
+            [self.hugo],
+            User.query.by_searchable_text(['Br', 'go']).all())
+
+    def test_by_searchable_text_ignores_empty_text(self):
+        self.assertEqual(
+            [self.jason, self.hugo, self.james],
+            User.query.by_searchable_text('').all())
```
Comment (deiferni) [inline opengever/ogds/models/tests/test_query.py]: I'd use an empty list/tuple here

**Comment by deiferni** [comment]
looks good :+1:

**Pull request merged**